src/filter/blur.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+import Image from '../image';
+
+// first release of mean filter
+export default function meanFilter(k){
+
+    if(this.components > 1){
```

Comment:
We have a method that allows to check this and that is called ["checkProcessable"](https://github.com/image-js/core/blob/master/src/image.js#L322-L349). 
We should take care that an image may have a bitDepth of 8, 16, 1, have alpha or not, have colorModel and many components (1 for grey, 3 for rgb, 4 for cmyk).
It is always better to be restrictive if we are not sure the code works in all the cases.
This method throws an error automatically if the requirements are not fulfilled.
In your case you should have something like:
this.checkProcessable({
components:[1],
bitDepth:[8,16]
})
A parameter that is not specified means that all the values are allowed.